### DISCUSSION TITLE
Fine-tuning

### DISCUSSION THREAD

**Discussion by botchi09** [oobabooga/text-generation-webui#238] [Q&A]
I find the API service really useful for running other applications. I'd like to fine-tune on some datasets I have (specifically for small models e.g. GPT2). Is there some way of getting it done in this? I remember I managed it a few years ago.

**Comment by oobabooga** [comment]
Hugging Face has some guides that might be useful as starting points:

https://huggingface.co/docs/transformers/training

**Reply by botchi09** [reply]
They were OK, but very temperemental. A script to load inside the repo might be handy.

**Comment by Satyam7166-tech** [comment]
@botchi09 , I don't know much about text genertion web UI.

Can you please tell me what you meant by the api service?

Right now I am using Ollama to for llama2 models. However I really need to fine tune it on my data. The thing is I am limited to using a mac studio (m2 ultra, 128gb memory, 24 core cpu and 76 core GPU) and the place that I work at don't want to host their data on cloud.

I can access my friend's gaming laptop though. However there is some CUDA installtion problem in his system.

Can you please help me out in this situation?
What are the options that I have and how I can approach fine tuning.

Thank you.
